mesheryctl/internal/cli/root/relationships/relationship.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2024 Layer5, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package relationships
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+	"github.com/manifoldco/promptui"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	outFormatFlag string
+
+	availableSubcommands = []*cobra.Command{ViewRelationshipsCmd}
+)
+
+var RelationshipCmd = &cobra.Command{
+	Use:   "relationships",
+	Short: "View list of relationships and detailed information of a specific relationship",
+	Long:  "View list of relationships and detailed information of a specific relationship",
+	Example: `
+// To view list of components
+mesheryctl exp relationships list
+
+// To view a specific model
+mesheryctl exp relationships view [model-name]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			utils.Log.Error(system.ErrGetCurrentContext(err))
+			return nil
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			utils.Log.Error(err)
```

Comment:
return err.

Reply:
We are logging the Error. Therefore i am returning nil.
Other wise we will have 2 same Error messages on the screen.

Reply:
Here we are logging the error message. Therefore, If we return the error there will 2 same error messages on the screen.
That's why i was returning nil.